lib/echocat.rb

```diff
@@ -1,7 +1,59 @@
 # frozen_string_literal: true
+# encoding: UTF-8
+
+require 'csv'
+require 'json'
 
 module Echocat
   def self.run
-    puts 'Hello World!'
+    csv_file_paths = [
+      '/workspaces/ruby-kata-1/data/authors.csv', 
+      '/workspaces/ruby-kata-1/data/books.csv', 
+      '/workspaces/ruby-kata-1/data/magazines.csv'
+    ]
+
+    convert_and_create_json_files(csv_file_paths)
+    find_by_isbn("5454-5587-3210")
+    find_by_authors_email("null-walter@echocat.org")
+    sort_by_title
+  end
+
+
+  def self.convert_and_create_json_files(csv_file_paths)
+    csv_file_paths.each do |csv_file_path|
+      csv_data = CSV.read(csv_file_path, headers: true, col_sep: ';', encoding: 'bom|utf-8')
+
+      json_data = csv_data.map { |row| row.to_h }
+
+      json_file_path = csv_file_path.sub('.csv', '.json')
+      
+      File.open(json_file_path, 'w') do |file|
+        file.puts JSON.pretty_generate(json_data)
+      end
+    end
+  end
+
+  def self.books_and_magazines
```

Comment:
Since this is called 3 times below it is opening the files and parsing the JSON 6 times all together. More reason to save the `json_data` into instance variables so you have them ready.